src/commons/api/ca.rs

```diff
@@ -1462,28 +1005,32 @@ impl From<&SigningCert> for ParentStatusIssuingCert {
 #[derive(Clone, Debug, Deserialize, Eq, PartialEq, Serialize)]
 pub struct ParentStatusCert {
     uri: uri::Rsync,
-    resources: ResourceSet,
     cert_pem: String,
 }
 
 impl From<&IssuedCert> for ParentStatusCert {
     fn from(issued: &IssuedCert) -> Self {
-        let cert = base64::encode(issued.cert.to_captured().as_slice());
+        let cert = base64::encode(issued.cert().to_captured().as_slice());
         let cert_pem = format!("-----BEGIN CERTIFICATE-----\n{}\n-----END CERTIFICATE-----\n", cert);
         ParentStatusCert {
             uri: issued.uri().clone(),
-            resources: issued.resource_set().clone(),
             cert_pem,
         }
     }
 }
 
-#[derive(Clone, Debug, Deserialize, Eq, PartialEq, Serialize)]
+#[derive(Clone, Debug, Default, Deserialize, Eq, PartialEq, Serialize)]
 pub struct ParentStatus {
     last_exchange: Option<ParentExchange>,
     last_success: Option<Timestamp>,
     all_resources: ResourceSet,
-    entitlements: HashMap<ResourceClassName, KnownEntitlement>,
+
+    // The struct changed - we did not record classes in 0.9.5 and below.
+    // Just default to an empty vec in case this field is missing, and
+    // ignore the 'entitlements' field that used to be there. This will
+    // be updated as soon as the CA synchronizes with its parent again.
+    #[serde(default)]
+    classes: Vec<ResourceClassEntitlements>,
```

Comment:
Why the change from HashMap to Vec? With a Vec you can now presumably get "duplicate" entries in the vec with the same `ResourceClassName` which before would not have been possible.

Reply:
The resource class name is part of `ResourceClassEntitlements`. It's true that duplicates could occur, but only if the parent sent us duplicates - which would be an error.